tests/test_derived_cameras.py

```diff
@@ -0,0 +1,351 @@
+        camera.distortion_params["k1"] = k1
+
+        # Warp the image
+        dewarped = cameras.warp_dewarp_image(
+            camera, gradient, warped_to_ideal=w2i, inversion_downsample=downsample
+        )
+
+        # Check whether the image on the average got lighter, darker, or
+        # stayed the same
+        assert dewarped.shape == gradient.shape
+        assert relationship(dewarped.mean(), gradient.mean())
+
+
+class TestPix2Face:
+
+    def test_simple_camera(self):
+        """Warp/dewarp should fail if not given a Metashape camera."""
+
+        # Create simple mesh and camera
+        mesh, point_colors = make_simple_mesh(pixels=[], color=None)
+        textured_mesh = TexturedPhotogrammetryMesh(
+            mesh=mesh,
+            input_CRS=EARTH_CENTERED_EARTH_FIXED_CRS,
+            texture=point_colors,
+        )
+        cameras = make_simple_camera_set()
+
+        # Render the mesh from these cameras
+        with pytest.raises(NotImplementedError):
+            textured_mesh.pix2face(
+                cameras=cameras,
+                cache_folder=None,
+                distortion_set=cameras,
+                apply_distortion=True,
+            )
+
+    @pytest.mark.parametrize("render_img_scale", [0.7, 0.8, 0.9, 1.0])
+    def test_dewarp_pix2face(self, tmp_path, render_img_scale):
+        """
+        Test that when we park a camera right over a mesh and call pix2face
+        the result gets warped (when set).
+        """
+
+        # Load a simple 2x2 mesh with no colors
+        mesh, point_colors = make_simple_mesh(pixels=[], color=None)
+        textured_mesh = TexturedPhotogrammetryMesh(
+            mesh=mesh,
+            input_CRS=EARTH_CENTERED_EARTH_FIXED_CRS,
+            texture=point_colors,
+        )
+
+        # Load a metashape camera set and then clean it up so it is looking
+        # directly at the simple mesh.
+        cameras = MetashapeCameraSet(
+            camera_file=camera_file(tmp_path),
+            image_folder=tmp_path,
+        )
+
+        # Pick an arbitrary sensor size and K1 such that the warping looks somewhat
+        # like our cameras do in practice (corners pulled slightly in).
+        # Note that simplify_camera zeroes out the other distortion parameters.
+        # Note that the choice of K1 is tied to the sensor size by the radius.
+        sensor = 2**8 + 1
+        camera = simplify_camera(cameras.cameras[0], image=np.ones((sensor, sensor, 3)))
+        camera.distortion_params["k1"] = -0.05
+
+        # In addition to simplifying the camera, we need to make sure the set transform
+        # is also removed
+        cameras._local_to_epsg_4978_transform = np.eye(4)
+
+        # Adjust the camera transform so we are looking down with the mesh face in view
+        HT = downward_view(
+            scene_width=4,  # Known width of the simple mesh in X/Y
+            focal=cameras.cameras[0].f,
```

Comment:
I see, I think the issue I mentioned about overwriting the camera.f value was related to doing this step first.